lib/packwerk/offenses_formatter.rb

```diff
@@ -1,12 +1,53 @@
 # typed: strict
 # frozen_string_literal: true
 
+require "packwerk/formatters/offenses_formatter"
+
 module Packwerk
   module OffensesFormatter
     extend T::Sig
     extend T::Helpers
 
-    interface!
+    abstract!
+
+    class << self
+      extend T::Sig
+
+      sig { params(base: Class).void }
+      def included(base)
+        @offenses_formatters ||= T.let(@offenses_formatters, T.nilable(T::Array[Class]))
+        @offenses_formatters ||= []
+        @offenses_formatters << base
+      end
+
+      sig { returns(T::Array[OffensesFormatter]) }
+      def all
+        load_default_formatter
+        T.unsafe(@offenses_formatters).map(&:new)
+      end
+
+      sig { params(identifier: String).returns(OffensesFormatter) }
+      def find(identifier)
+        formatter_by_identifier(identifier)
+      end
+
+      private
+
+      sig { void }
+      def load_default_formatter
```

Comment:
There isn't a need to referencer the constant if we require the offense formatter at the top of the file. If the formatter is only used when checking, maybe the best thing to do would be to do a runtime `require "packwerk/formatters/offenses_formatter"` in this function?

Reply:
Sure, I like that! Definitely clearer that we're explicitly loading rather than loading through invoking auto load.